checkov/terraform/checks/graph_checks/gcp/GCPPostgreSQLDatabaseFlaglog_statement_statsIsSetToOFF.yaml

```diff
@@ -0,0 +1,23 @@
+metadata:
+  name: "Ensure PostgreSQL database flag 'log_statement_stats' is set to 'off'"
+  id: "CKV2_GCP_17"
+  category: "GENERAL"
+definition:
+  and:
+    - cond_type: "attribute"
+      resource_types:
+        - "google_sql_database_instance"
+      attribute: "database_version"
+      operator: "contains"
+      value: "POSTGRES"
+    - or:
+        - cond_type: "attribute"
+          resource_types:
+            - "google_sql_database_instance"
+          attribute: "settings.database_flags[*]"
+          operator: "jsonpath_not_exists"
+        - cond_type: "attribute"
+          resource_types:
+            - "google_sql_database_instance"
+          attribute: "settings.database_flags[?(@.name == log_statement_stats & @.value == on)]"
+          operator: "jsonpath_not_exists"
```

Comment:
same here, one big `or:`

Reply:
CKV2_GCP_13 - has one big `or:` and the rest 4 Database policies has `and:`  & `or:` because,

By default, GCP turns **off** these flags, so these flags will be turned off when not used in the script.
https://cloud.google.com/sql/docs/postgres/flags

So we have three cases to be satisfied,

1. It should be a PostgreSQL instance
2. The corresponding flag attribute should not be part of the settings
3. If the flag is present, then it should be turned OFF

That's why I have mentioned `or:` & `and:` for the policies where flags need to be turned **OFF**
Let me know if you have a different logic.

Reply:
I have updated the suggested changes.